Sources/MapboxNavigation/SpriteRepository.swift

```diff
@@ -0,0 +1,147 @@
+import UIKit
+import MapboxMaps
+import MapboxCoreNavigation
+import MapboxDirections
+
+class SpriteRepository {
+    let imageCache: BimodalImageCache
+    let metadataCache =  SpriteMetaDataCache()
+    var styleURI: StyleURI = .navigationDay
+    var baseURL: URL = URL(string: "https://api.mapbox.com/styles/v1")!
+    
+    public var sessionConfiguration: URLSessionConfiguration = URLSessionConfiguration.default {
+        didSet {
+            imageDownloader = ImageDownloader(sessionConfiguration: sessionConfiguration)
+        }
+    }
+
+    public static let shared = SpriteRepository.init()
+    fileprivate(set) var imageDownloader: ReentrantImageDownloader
+    
+    init(imageCache: BimodalImageCache = ImageCache(), withDownloader downloader: ReentrantImageDownloader = ImageDownloader()) {
+        self.imageCache = imageCache
+        self.imageDownloader = downloader
+    }
+    
+    func updateRepository(styleURI: StyleURI? = nil, shield: VisualInstruction.Component.ShieldRepresentation? = nil, completion: @escaping CompletionHandler) {
+        let baseURL = shield?.baseURL ?? self.baseURL
+        let styleURI = styleURI ?? self.styleURI
+        
+        guard let styleID = styleURI.rawValue.components(separatedBy: "styles")[safe: 1],
+              let spriteRequestURL = spriteURL(isImage: true, baseURL: baseURL, styleID: styleID),
+              let metadataRequestURL = spriteURL(isImage: false, baseURL: baseURL, styleID: styleID) else {
+                  return
+              }
+
+        let dispatchGroup = DispatchGroup()
+        
+        dispatchGroup.enter()
+        downloadMetadata(metadataRequestURL) { (_) in
+            dispatchGroup.leave()
+        }
+        
+        dispatchGroup.enter()
+        downLoadSprite(spriteRequestURL) { (_) in
+            dispatchGroup.leave()
+        }
+        
+        dispatchGroup.notify(queue: .main) {
+            self.styleURI = styleURI
+            self.baseURL = baseURL
+            completion()
+        }
+    }
+    
+    func spriteURL(isImage: Bool, baseURL: URL, styleID: String) -> URL? {
+        guard var urlComponent = URLComponents(url: baseURL, resolvingAgainstBaseURL: false),
+              let accessToken = NavigationSettings.shared.directions.credentials.accessToken else { return nil }
+        
+        let requestTyep = isImage ? "/sprite@2x.png" : "/sprite@2x"
```

Comment:
Typo: `requestType`.

Reply:
Should we be hardcoding 2× resolution here?

Reply:
I'll use the` VisualInstruction.Component.scale` instead.